scene/animation/animation_node_state_machine.cpp

```diff
@@ -193,8 +193,20 @@ AnimationNodeStateMachineTransition::AnimationNodeStateMachineTransition() {
 
 ////////////////////////////////////////////////////////
 
+void AnimationNodeStateMachinePlayback::_set_fading_from(const StringName &p_fading_from) {
+	fading_from = p_fading_from;
+
+	emit_signal(SNAME("fading_from_changed"), fading_from);
+}
+
 void AnimationNodeStateMachinePlayback::_set_current(AnimationNodeStateMachine *p_state_machine, const StringName &p_state) {
 	current = p_state;
+	if (is_grouped) {
+		_get_parent_playback(p_state_machine->get_animation_tree())->emit_signal(SNAME("current_changed"), p_state);
+	} else {
```

Comment:
Nesting can be multi-layered, so it needs to add like the following method to parent StateMachine to call it:

```cpp
void current_state_changed_in_child(String p_left_over) {
	String full_path = current + "/" + p_left_over
	if (is_grouped) {
		_get_parent_playback(p_state_machine->get_animation_tree())->current_state_changed_in_child(full_path);
	} else {
		emit_signal(SNAME("current_changed"), full_path);
	}
}
```